### PR TITLE
Changes according to commit message

### PR DESCRIPTION
**Pull request by hans99**
This is what I am running now.

### CODE REVIEW

File: usb.cpp [hunk 1]
```diff
@@ -33,23 +33,24 @@ USB::received(struct libusb_transfer* transfer)
...
-    {
-	fprintf(stderr, "irq transfer status %d, terminating\n", transfer->status);
-
-	do_exit = 2;
-	libusb_free_transfer(transfer);
-	recv_transfer = NULL;
-    }
-    else
-    {
+	// This doesnt work for me
```
Comment (karloygard) [inline usb.cpp]: Why did you have to disable this--did it just not work, or what happened?

File: main.cpp [hunk 2]
```diff
@@ -266,6 +266,49 @@ void MQTTGateway::mqtt_connect(mosquitto* mosq, void* obj, int rc)
...
 }
 
+void MQTTGateway::mqtt_disconnect(mosquitto* mosq, void* obj, int rc)
+{
+    printf("MQTT, Disconnect (%d)\n", rc);
+    //syslog(LOG_INFO, "MQTT, Disconnect (%d)\n", rc);
+
+    if (rc)
+    {
+        sleep(1);
```
Comment (karloygard) [inline main.cpp]: Do you need the sleep() here?

File: usb.cpp (hunk 1 above)
Comment (hans99) [reply usb.cpp]: I get error first thing but still works, will get more info ...

File: main.cpp (hunk 2 above)
Comment (hans99) [reply main.cpp]: Simple way to minimize impact of possible disconnect/connect race, we're in trouble now anyways ... but I know :)

Comment (karloygard) [reply main.cpp]: Aha, makes sense.  I'll play a little with it and see what I come up with.

**Pull request closed without merging**